analysis.R

```diff
@@ -1,138 +1,115 @@
-###################
-##  12/27/21  ##
-###################
+#####################################
+##  Depression-Belief Interaction  ##
+#####################################
 
 ####  Startup  ####
-
 # Load packages
-
-if(!require(tidyverse)){install.packages('tidyverse')}
-if(!require(here)){install.packages('here')}
-if(!require(ggplot2)){install.packages('ggplot2')}
-
 library(tidyverse)
 library(here)
 library(ggplot2)
 
 
-#### Load data ####
-
-# The here() package returns filepaths based on the arguments you give it,
-# starting from the folder with your .Rproj file; in this case, it will return
-# the filepath to the dataset inside the "data" folder
-# This is especially useful when a script will be run on Windows and Mac 
-# computers, as they use different folder separators ("/" vs "\")
 
+#### Load data ####
 df_1 <- readRDS(here("data", "T2T Data for Jans et al.rds"))
 
 
-####  Prepare Data  ####
 
+####  Prepare Data  ####
+df_2 <- df_1 %>%
+  
   # Adding a column for CDI mean
-
-  df_2 <- df_1 %>% mutate(cdi_mean = rowMeans(select(.,contains("cdi")), na.rm = FALSE))
+  mutate(cdi_mean = rowMeans(select(.,contains("cdi")), na.rm = FALSE)) %>%
   
   # Selecting variables
-  
-  df_3 <- df_2 %>% select(yb_lsmh_id, yb_permanence, yb_cause_brain, yb_cause_env, yb_change_brain, yb_change_env, cdi_mean)
+  select(yb_lsmh_id,
+         # Demographic variables
+         pb_childgender, pb_income, pb_childethnicity,
+         # Variables for analysis
+         yb_permanence, yb_cause_brain, yb_cause_env, yb_change_brain, yb_change_env, cdi_mean) %>%
   
   # Removing rows containing NA
+  filter(complete.cases(.)) %>%
   
-  df_4 <- df_3 %>% filter(complete.cases(.))
-  
+  # Standardize numeric variables
+  mutate(across(c(yb_permanence, yb_cause_brain, yb_cause_env, yb_change_brain, yb_change_env, cdi_mean), 
+                scale,
+                .names = "{col}_scaled"))
+
+
 
 ####  Analysis  ####
 
 ## Regression with CDI as covariate
 
-my_model_1 <- lm(data = df_4, yb_permanence ~ yb_cause_brain + yb_change_brain + yb_cause_brain*yb_change_brain + cdi_mean)
+my_model_1 <- lm(data = df_2, yb_permanence_scaled ~ yb_cause_brain_scaled + yb_change_brain_scaled + yb_cause_brain_scaled*yb_change_brain_scaled + cdi_mean_scaled)
```

Comment:
It's important to scale your variables when you're doing anything with an interaction. I also find it easier to interpret generally. Basically, by scaling the variables they're now in terms of standard deviations (a value of 0 is equal to the mean of the original variable, a value of 1 is equal to one standard deviation above the mean of the original value, etc.). This makes it easier to interpret regression coefficients when your variables don't have naturally meaningful levels since you can say "a 1-standard deviation increase in X is associated with a _-standard deviation change in Y"